### PR TITLE
update  getEndpoint() for us gov urls

### PR DESCRIPTION
**Pull request by tomlm**
Fixes

### CODE REVIEW

File: libraries/botframework-config/src/models/luisService.ts [hunk 1]
```diff
@@ -48,7 +48,17 @@ export class LuisService extends ConnectedService implements ILuisService {
 
     // get endpoint for the luis service
     public getEndpoint(): string {
-        return `https://${this.region}.api.cognitive.microsoft.com`;
+        let reg  = this.region.toLowerCase(); 
+        if (reg == "virginia" || reg == "usgovvirginia")
```
Comment (benbrown) [inline libraries/botframework-config/src/models/luisService.ts]:
```suggestion
        if (reg === "virginia" || reg === "usgovvirginia") 
```

File: libraries/botframework-config/src/models/luisService.ts [hunk 2]
```diff
@@ -48,7 +48,17 @@ export class LuisService extends ConnectedService implements ILuisService {
...
     public getEndpoint(): string {
-        return `https://${this.region}.api.cognitive.microsoft.com`;
+        let reg  = this.region.toLowerCase(); 
+        if (reg == "virginia" || reg == "usgovvirginia") 
+        { 
+            return `https://virginia.api.cognitive.microsoft.us`; 
+        } 
+        else if (reg.startsWith("usgov") || reg.startsWith("usdod")) 
+        { 
+            return `https://${this.region}.api.cognitive.microsoft.us`;
```
Comment (carlosscastro) [inline libraries/botframework-config/src/models/luisService.ts]: Add a comment on why we do this? It is not obvious :)

File: libraries/botframework-config/src/models/luisService.ts [hunk 3]
```diff
@@ -48,7 +48,17 @@ export class LuisService extends ConnectedService implements ILuisService {
...
+        if (reg == "virginia" || reg == "usgovvirginia") 
+        { 
+            return `https://virginia.api.cognitive.microsoft.us`; 
+        } 
+        else if (reg.startsWith("usgov") || reg.startsWith("usdod")) 
+        { 
+            return `https://${this.region}.api.cognitive.microsoft.us`; 
+        } 
+ 
+        return `https://${this.region}.api.cognitive.microsoft.com`;
```
Comment (benbrown) [inline libraries/botframework-config/src/models/luisService.ts]:
```suggestion
        return `https://${ this.region }.api.cognitive.microsoft.com`;     
```

File: libraries/botframework-config/tests/service.test.js [hunk 4]
```diff
@@ -8,6 +8,15 @@ describe("Service Tests", () => {
     it("Luis.getEndpoint() returns correct url for region", async () => {
         let luis = new bf.LuisService({ region: "westus" });
         assert.equal(luis.getEndpoint(), `https://westus.api.cognitive.microsoft.com`);
+
+        luis = new bf.LuisService({ region: "virginia" });
```
Comment (carlosscastro) [inline libraries/botframework-config/tests/service.test.js]: Comment on this and why it is there? New contributors may not know why virginia is special, and that it is related to government.

File: libraries/botframework-config/src/models/luisService.ts (hunk 3 above)
Comment (cleemullins) [reply libraries/botframework-config/src/models/luisService.ts]: Linting for the win! :)

Comment (tomlm) [reply libraries/botframework-config/src/models/luisService.ts]: Why?

**Pull request merged**